axum-macros/src/from_request.rs

```diff
@@ -0,0 +1,186 @@
+
+fn extract_fields<'a, I>(fields: I) -> syn::Result<Vec<TokenStream>>
+where
+    I: Iterator<Item = &'a syn::Field>,
+{
+    fields
+        .enumerate()
+        .map(|(index, field)| {
+            let FromRequestAttrs { via } = parse_attrs(&field.attrs)?;
+
+            let member = if let Some(ident) = &field.ident {
+                quote! { #ident }
+            } else {
+                let member = syn::Member::Unnamed(syn::Index {
+                    index: index as u32,
+                    span: field.span(),
+                });
+                quote! { #member }
+            };
+
+            let ty_span = field.ty.span();
+
+            let from_request_ty = if let Some(via) = &via {
+                let span = via.span();
+                quote_spanned! {span=>
+                    <#via<_> as ::axum::extract::FromRequest<B>>
+                }
+            } else {
+                quote_spanned! {ty_span=>
+                    ::axum::extract::FromRequest
+                }
+            };
+
+            let into_inner = if let Some(via) = via {
+                let span = via.span();
+                quote_spanned! {span=>
+                    |#via(inner)| inner
+                }
+            } else {
+                quote_spanned! {ty_span=>
+                    ::std::convert::identity
+                }
+            };
+
+            Ok(quote_spanned! {ty_span=>
+                #member: {
+                    #from_request_ty::from_request(req)
+                        .await
+                        .map(#into_inner)
+                        .map_err(::axum::response::IntoResponse::into_response)?
+                },
+            })
+        })
+        .collect()
+}
+
+#[derive(Debug, Default)]
+struct FromRequestAttrs {
+    via: Option<syn::Path>,
+}
+
+fn parse_attrs(attrs: &[syn::Attribute]) -> syn::Result<FromRequestAttrs> {
+    #[derive(Debug)]
+    enum Attr {
+        FromRequest(Punctuated<FromRequestAttr, Token![,]>),
+    }
+
+    #[derive(Debug)]
+    enum FromRequestAttr {
+        Via { via: kw::via, path: syn::Path },
+    }
+
+    impl Parse for FromRequestAttr {
+        fn parse(input: ParseStream) -> syn::Result<Self> {
+            let lh = input.lookahead1();
+            if lh.peek(kw::via) {
+                let via = input.parse::<kw::via>()?;
+                let content;
+                syn::parenthesized!(content in input);
+                content.parse().map(|path| Self::Via { via, path })
+            } else {
+                Err(lh.error())
+            }
+        }
+    }
+
+    mod kw {
+        syn::custom_keyword!(via);
+    }
+
+    let attrs = attrs
+        .iter()
+        .filter_map(|attr| attr.path.get_ident().map(|ident| (ident, attr)))
+        .map(|(ident, attr)| {
+            if ident == "from_request" {
+                attr.parse_args_with(Punctuated::parse_terminated)
+                    .map(Attr::FromRequest)
+            } else {
```

Comment:
If there are attributes other than `from_request`, they should simply be ignored rather than raising an error. They could be for a different macro (or builtin, though cfg attrs are expanded before derive macro expansion).